file/validate.go

```diff
@@ -29,5 +31,86 @@ func validate(content []byte) error {
 		err := fmt.Errorf(desc.String())
 		errs.Errors = append(errs.Errors, err)
 	}
+
+	var kongdefaults *KongDefaults
+	if err := json.Unmarshal(content, kongdefaults); err != nil {
+		return fmt.Errorf("unmarshaling file into KongDefaults: %w", err)
+	}
+	if kongdefaults != nil {
+		if err := checkDefaults(*kongdefaults); err != nil {
+			return fmt.Errorf("default values are not allowed for these fields: %w", err)
+		}
+	}
 	return errs
 }
+
+func check(item string, t *string) string {
+	if t != nil && len(*t) > 0 {
+		return item
+	}
+	return ""
+}
+
+func checkDefaults(kd KongDefaults) error {
+	var invalid []string
+	if kd.Service == nil {
+		invalid = append(invalid, "Invalid Service Address")
```

Comment:
This is not an error because the user can choose to not specify a default for service.
If service is nil, you can skip checking any of the properties all together and that's it.

The same comment applies for route, target and upstream.

Reply:
changed 507639081e5cb99ce9e720a28e58c17b7afdafc4